src/BeforeMiddlewareCaller.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace DI\Bridge\Silex;
+
+use DI\Bridge\Silex\Application;
+use Invoker\Reflection\CallableReflection;
+use Invoker\ParameterResolver\ResolverChain;
+use Invoker\ParameterResolver\Container\TypeHintContainerResolver;
+use Invoker\ParameterResolver\NumericArrayResolver;
+use Invoker\ParameterResolver\ParameterResolver;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class that takes the responsibility of resolving and calling a before middleware with injection
+ *
+ * @author Felix Becker <f.becker@outlook.com>
+ */
+class BeforeMiddlewareCaller
```

Comment:
The difference is that after middleware gets passed (request, response, app) while before middleware gets passed (request, app), because the response doesn't exist yet. I feel like making them inherit from an abstract base class and refactoring few lines of code into common methods would result in more code than it would save